Source/DafnyCore/Compilers/SinglePassCompiler.cs

```diff
@@ -1335,7 +1335,7 @@ public enum ResolvedUnaryOp { BoolNot, BitwiseNot, Cardinality }
           } else if (d is IteratorDecl) {
             var iter = (IteratorDecl)d;
             if (DafnyOptions.O.ForbidNondeterminism && iter.Outs.Count > 0) {
-              Error(iter.tok, "since yield parameters are initialized arbitrarily, iterators are forbidden by /definiteAssignment:3 option", wr);
+              Error(iter.tok, "since yield parameters are initialized arbitrarily, iterators are forbidden by enforce-determinism option", wr);
```

Comment:
```suggestion
              Error(iter.tok, "since yield parameters are initialized arbitrarily, iterators are forbidden by the --enforce-determinism option", wr);
```